monkey/monkey_island/cc/server_utils/encryption/data_store_encryptor.py

```diff
@@ -45,12 +45,26 @@ def _create_key(self) -> KeyBasedEncryptor:
 
         return KeyBasedEncryptor(plaintext_key)
 
-    def encrypt(self, plaintext: str) -> str:
+    def encrypt(self, plaintext: Any) -> Any:
+        if isinstance(plaintext, dict):
+            return DataStoreEncryptor._dictionary_encryptor(
+                plaintext, self._key_based_encryptor.encrypt
+            )
         return self._key_based_encryptor.encrypt(plaintext)
 
-    def decrypt(self, ciphertext: str):
+    def decrypt(self, ciphertext: Any):
+        if isinstance(ciphertext, dict):
+            return DataStoreEncryptor._dictionary_encryptor(
+                ciphertext, self._key_based_encryptor.decrypt
+            )
         return self._key_based_encryptor.decrypt(ciphertext)
 
+    @staticmethod
+    def _dictionary_encryptor(
+        plaindictionary: Mapping[str, str], encryptor: Callable[[str], str]
+    ) -> Mapping[str, str]:
+        return {item_key: encryptor(item_value) for item_key, item_value in plaindictionary.items()}
+
```

Comment:
Use `monkey_island.cc.server_utils.encryptions.dict_encryptor` instead of making these changes.